pkg/network/usm/ebpf_main.go

```diff
@@ -354,6 +360,30 @@ func (e *ebpfProgram) init(buf bytecode.AssetReader, options manager.Options) er
 	return e.InitWithOptions(buf, options)
 }
 
+const connProtoTTL = 3 * time.Minute
+const connProtoCleaningInterval = 5 * time.Minute
+
+func (e *ebpfProgram) setupMapCleaner() *ddebpf.MapCleaner {
+	mapCleaner, err := ddebpf.NewMapCleaner(e.connectionProtocolMap, new(netebpf.ConnTuple), new(netebpf.ProtocolStackWrapper))
+	if err != nil {
+		log.Errorf("error creating map cleaner: %s", err)
+		return nil
+	}
```

Comment:
return the error to the caller and let it decide what to do with it

Reply:
This is a private method with one caller, but sure I can move the error handling 🤷

Reply:
the pattern is the problem, of returnning nil objects and that should indicate the error while errors are silently logged